eclair-core/src/main/scala/fr/acinq/eclair/db/FileBackupHandler.scala

```diff
@@ -16,80 +16,100 @@
-        val tmpFile = new File(backupFile.getAbsolutePath.concat(".tmp"))
-        databases.backup(tmpFile)
+  /**
+   * @param targetFile  backup file
+   * @param script_opt  (optional) script to execute after the backup completes
+   * @param interval    interval between two backups
+   */
+  case class FileBackupParams(interval: FiniteDuration,
+                              targetFile: File,
+                              script_opt: Option[String])
 
-        // this will throw an exception if it fails, which is possible if the backup file is not on the same filesystem
-        // as the temporary file
-        Files.move(tmpFile.toPath, backupFile.toPath, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE)
+  sealed trait Command
+  case class WrappedChannelPersisted(wrapped: ChannelPersisted) extends Command
+  private case object TickBackup extends Command
 
-        // publish a notification that we have updated our backup
-        context.system.eventStream.publish(BackupCompleted)
-        Metrics.FileBackupCompleted.withoutTags().increment()
-      }
+  sealed trait BackupEvent
+  // this notification is sent when we have completed our backup process (our backup file is ready to be used)
+  case object BackupCompleted extends BackupEvent
+  // @formatter:on
 
-      backupScript_opt.foreach(backupScript => {
-        Try {
-          // run the script in the current thread and wait until it terminates
-          Process(backupScript).!
-        } match {
-          case Success(exitCode) => log.debug(s"backup notify script $backupScript returned $exitCode")
-          case Failure(cause) => log.warning(s"cannot start backup notify script $backupScript:  $cause")
-        }
-      })
-  }
+  def apply(databases: FileBackup, backupParams: FileBackupParams): Behavior[Command] =
+    Behaviors.setup { context =>
+      // we listen to ChannelPersisted events, which will trigger a backup
+      context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelPersisted](WrappedChannelPersisted))
+      Behaviors.withTimers { timers =>
+        timers.startTimerAtFixedRate(TickBackup, backupParams.interval)
+        new FileBackupHandler(databases, backupParams, context).normal(false)
+      }
+    }
 }
 
-sealed trait BackupEvent
+class FileBackupHandler private(databases: FileBackup,
+                                backupParams: FileBackupParams,
+                                context: ActorContext[Command]) {
 
-// this notification is sent when we have completed our backup process (our backup file is ready to be used)
-case object BackupCompleted extends BackupEvent
+  def normal(willBackup: Boolean): Behavior[Command] =
+    Behaviors.receiveMessagePartial {
+      case _: WrappedChannelPersisted =>
+        context.log.debug("will perform backup")
+        normal(willBackup = true)
+      case TickBackup => if (willBackup) {
+        context.log.debug("performing backup")
+        doBackup()
```

Comment:
IIRC we had a bounded mailbox because `doBackup` can be slow, during which time the mailbox is getting full (if there is a lot of channel churn) which can cause issues. Do we want to do something about it? Or should we just direct users with enough load towards postgres which disables this component?

Reply:
That's a good point, I knew I was missing something...

Reply:
Fixed with d2a3998fc.